### PR TITLE
Fix ASPP ONNX export by pre-building ResizingND in build()

### LINKED ISSUE #89: AtrousSpatialPyramidPooling: dynamic ResizingND in call() breaks ONNX export
**Describe the bug**

`AtrousSpatialPyramidPooling.call()` creates a new `ResizingND` on every forward pass with a dynamic target shape from `ops.shape(inputs)`. ONNX exporters trace the graph once and cannot handle new layer objects during tracing. This makes `DeepLabV3Plus` the only medicai segmentation model that cannot be exported to ONNX.

**To Reproduce**

```python
from medicai.models import DeepLabV3Plus
import numpy as np

model = DeepLabV3Plus(input_shape=(64, 64, 3), num_classes=2, encoder_name="efficientnet_v2_s", encoder_depth=3)
model(np.zeros((1, 64, 64, 3), dtype="float32"))
model.export("model.onnx", format="onnx")
```

**Cause**

In `medicai/layers/conv.py`, `AtrousSpatialPyramidPooling.call()`:

```python
input_shape = ops.shape(inputs)
target_size = tuple(input_shape[i] for i in range(1, 1 + self.spatial_dims))
result[-1] = ResizingND(target_size, ...)(result[-1])
```

The target spatial dimensions are statically known at `build()` time. The fix is to pre-build the `ResizingND` in `build()` and reuse it in `call()`. No change to weights, architecture, or numerical output.

**Environment**
- medicai 0.0.3
- Keras 3.13.2, PyTorch 2.10.0
- onnx 1.21.0, onnxscript 0.6.2
- Python 3.14.3

### PR DESCRIPTION
**Pull request by linusjuni**
Fixes #89

`AtrousSpatialPyramidPooling.call()` creates a new `ResizingND` on every forward pass, which breaks ONNX export. The target spatial dimensions are statically known at `build()` time, so this moves the `ResizingND` instantiation from `call()` to `build()`.

Also adds a `test_deeplabv3plus` test (2D + 3D) to match the other segmentation model tests.

## Summary by CodeRabbit

* **New Features**
  * DeepLabV3Plus model now available with support for both 2D and 3D configurations

* **Tests**
  * Added test coverage for DeepLabV3Plus across multiple dimensional configurations

### CODE REVIEW

**Comment by linusjuni** [comment]
Thanks for building medicai - it's been really useful for our imaging work. This is my first contribution to the repo, happy to adjust anything if needed!

**Comment by linusjuni** [comment]
**Regarding the AI code-reviews**

Creating a layer inside `call()` every forward pass is a Keras anti-pattern - it breaks serialization, weight tracking, JAX tracing, and graph-based exports. Moving it to `build()` is the standard fix. 

ASPP's `build()` always gets concrete int shapes - no medicai model passes None spatial dims. Calling a model with a different spatial size after building would break everything (batch norms, convs, concat), not just the resize.

The test follows the same pattern as all other model tests in the repo.

**Comment by innat** [comment]
@linusjuni 
Thanks for the PR. This is indeed a bug. Will take look soon.

File: test/models/test_models.py [hunk 1]
```diff
@@ -212,6 +213,34 @@ def test_vit():
...
+    input_shape = (64, 64, 3)
+    model = DeepLabV3Plus(
+        input_shape=input_shape,
+        num_classes=num_classes,
+        encoder_name="efficientnet_v2_s",
+        encoder_depth=3,
+    )
+    dummy_input = tf.random.normal((1, 64, 64, 3))
+    output = model(dummy_input)
+    assert output.shape == (1, 64, 64, num_classes)
```
[the remark this answers is not in the document]
Comment (innat) [reply test/models/test_models.py:241]: @linusjuni 
You don't need to add any test case. Current test case is quite fragile, will be updated soon.

File: medicai/layers/conv.py [hunk 2]
```diff
@@ -550,6 +550,18 @@ def build(self, input_shape):
...
+        pool_target = tuple(input_shape[i] for i in range(1, 1 + spatial_dims))
+        self._pool_resize = ResizingND(
+            target_shape=pool_target,
+            interpolation="bilinear" if spatial_dims == 2 else "trilinear",
+            name="pool_resize",
+        )
+        pool_resize_input = (
+            input_shape[:1] + (1,) * spatial_dims + (self.num_channels,)
+        )
+        self._pool_resize.build(pool_resize_input)
```
[the remark this answers is not in the document]
Comment (innat) [reply medicai/layers/conv.py:563]: @linusjuni 
Can you try following the above suggestion and report how it goes?

[the remark this answers is not in the document]
Comment (linusjunigubra) [reply medicai/layers/conv.py:563]: @innat
Yes sure!😄

`ops.shape(inputs)` in `call()` returns symbolic tensors during graph tracing, not concrete ints - that's the bug this PR fixes. Using `ops.image.resize` / `resize_volumes` with `ops.shape()` would bring back the same problem.

Dynamic shapes shouldn't be a concern here because `build()` always gets concrete ints since all models need a concrete `input_shape`. `ResizingND.build()` already does the same thing for JAX (pre-computes `computed_target_shape`) for the same reason.

Happy to explore alternatives if you'd prefer a different approach though.
